samigo/samigo-api/src/java/org/sakaiproject/tool/assessment/data/dao/grading/ItemGradingData.java

```diff
@@ -21,299 +21,53 @@
 
 package org.sakaiproject.tool.assessment.data.dao.grading;
 
-import java.util.ArrayList;
 import java.util.Date;
 import java.util.HashSet;
-import java.util.Iterator;
 import java.util.List;
 import java.util.Set;
 
-import org.apache.commons.lang3.builder.EqualsBuilder;
-import org.apache.commons.lang3.builder.HashCodeBuilder;
-import lombok.extern.slf4j.Slf4j;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
 
 /**
  * A response to a specific question and its associated data
- * 
- * 
  */
-@Slf4j
+@Getter
+@Setter
+@EqualsAndHashCode
```

Comment:
I think the equals and hashcode are incorrect and need to better identify equality.

Reply:
Why? No point doing the call to super - it's just Object, unless you're saying there's a proxy we need to account for? Calling super on an AOP generated proxy.

Reply:
Do you mean the original code was bad, too.

Reply:
When you add lombok annotation it will use every non transient, non static fields in the equals and hashcode implementation. The version you replaced didn't use every field in the equals and hashcode implementation which is why I mention this.

You likely need to look at using:
@EqualsAndHashCode.Include or @EqualsAndHashCode.Exclude

Reply:
Fixed it. Took the list out.